schema-registry/core/src/main/java/com/hortonworks/registries/schemaregistry/ISchemaRegistry.java

```diff
@@ -82,6 +82,15 @@ default SchemaVersionInfo getSchemaVersionInfo(String schemaName, String schemaT
      */
     SchemaVersionInfo getSchemaVersionInfo(String schemaName, String schemaText, boolean disableCanonicalCheck) throws SchemaNotFoundException, InvalidSchemaException, SchemaBranchNotFoundException;
 
+    /**
+     * If there is a version of the schema with the given fingerprint then it returns the respective {@link SchemaVersionInfo}, else it returns null.
```

Comment:
Instead of "else it returns null" can we replace it with "else it throws SchemaNotFoundException"?